test/distributed/org/apache/cassandra/distributed/test/ring/BootstrapTest.java

```diff
@@ -95,6 +95,32 @@ public void autoBootstrapTest() throws Throwable
                 Assert.assertEquals("Node " + e.getKey() + " has incorrect row state", e.getValue().longValue(), 100L);
         }
     }
+    
+    @Test
+    public void hostIdOverrideTest() throws Throwable
```

Comment:
This is looking great, good job! I think we can still improve it:
a) check that original node contains host id of second node on `system.peers` table
b) after the node starts with the provided `host_id_first_boot`, stop the node and start with a different `host_id_first_boot` - check the id should not change since only the first one should be used.
c) what happens when `host_id_first_boot` is invalid? node startup should fail with error message and we should test this

Reply:
I have done two first points.

About last one.If I provide invalid host_id, cassandra throws an exception and does not start.So, how can I catch that exception?

Reply:
added additional test with assertion

Reply:
hey @fibersel , if something is done, just click the "resolved" button, it just makes less visual clutter ...